builtin/k8s/releaser.go

```diff
@@ -339,10 +353,366 @@ func (r *Releaser) resourceServiceDestroy(
 		return err
 	}
 
+	step.Update("Service deleted")
 	step.Done()
 	return nil
 }
 
+func (r *Releaser) resourceIngressCreate(
+	ctx context.Context,
+	log hclog.Logger,
+	target *Deployment,
+	result *Release,
+	state *Resource_Ingress,
+	serviceState *Resource_Service,
+	csinfo *clientsetInfo,
+	sg terminal.StepGroup,
+) error {
+	// Preflight config checks
+	if r.config.IngressConfig != nil {
+		if r.config.LoadBalancer {
+			return status.Error(codes.FailedPrecondition, "A LoadBalancer service type is not "+
+				"compatible with an Ingress config. Please pick one or the other for your release")
+		}
+		if r.config.NodePort != 0 {
+			return status.Error(codes.FailedPrecondition, "A NodePort service type is not "+
+				"compatible with an Ingress config. Please pick one or the other for your release")
+		}
+
+		if r.config.IngressConfig.ClassName != "http" {
+			return status.Error(codes.FailedPrecondition, "An ingress stanza must be "+
+				"of type \"http\".")
+		}
+	} else {
+		// No ingress config was set, we're not configuring an ingress resource
+		return nil
+	}
```

Comment:
Little "boolean Zen" thing here but I would just change this whole thing to:

```go
if r.config.IngressConfig == nil {
    return nil
}

// Do checks
```

So you can lower the overall depth of this by one.